actors/builtin/miner/expiration_queue.go

```diff
@@ -698,3 +716,126 @@ func groupSectorsByExpiration(sectorSize abi.SectorSize, sectors []*SectorOnChai
 	})
 	return sectorEpochSets
 }
+
+// Groups sectors into sets based on their Expiration field.
+// If sectors are not found in the expiration set corresponding to their expiration field
+// (i.e. they have been rescheduled) traverse expiration sets to for groups where these
+// sectors actually expire.
+// Groups will be returned in expiration order, earliest first.
+func (q *ExpirationQueue) findSectorsByExpiration(sectorSize abi.SectorSize, sectors []*SectorOnChainInfo) ([]sectorExpirationSet, error) {
+	declaredExpirations := make(map[abi.ChainEpoch]bool, len(sectors))
+	sectorsByNumber := make(map[uint64]*SectorOnChainInfo, len(sectors))
+	emptyBitfield := bitfield.New()
+	allNotFound := &emptyBitfield
```

Comment:
We changed to pass Bitfields by value type a month or so back. Change`groupExpirationSet` to return a non-pointer to bitfield and remove the pointers here.

But also consider just using a `map[uint64]struct{}`, which will save the underlying unnecessary RLE

Reply:
It's a little sad we can't efficiently do proper set math on bitfields. I spent some time trying to figure out how this could work. If we could copy `RunIterators`, we could use runs to perform multiple set operations and then convert back to bitfields efficiently (if necessary).

Using a map here ended up being less code anyway, so I went ahead and made that optimization.